### PR TITLE
Fix incorrect SuperDSC backGap assignment for GQA-style folded views under coarse tiling.

### LINKED ISSUE #3098: Coarse `Lq` tiling produces wrong results for GQA flash attention with folded `G * Lq` dimension
### What happened?

**Description**

`flash_spyre` in `test-spyre-scripts/test_flash_gqa.py` produces correct results without an `Lq` tiling hint, but mismatches when enabling:

```python
with spyre_hint(tiles={"Lq": Lq // q_block_size}):
```

The implementation folds queries from `[B, Hq, Lq, D]` into `[B, Hkv, G * Lq, D]` for GQA, with `Hq = Hkv * G`.

Without the `Lq` hint, the test passes. With the hint enabled around the attention body, the result mismatches:

```text
Mismatched elements: ~1,017,611 / 16,777,216
Greatest absolute difference: ~3.49
```

`flash_cpu` was also checked against:

```python
torch.nn.functional.scaled_dot_product_attention(..., enable_gqa=True)
```

and matched within tolerance, so this does not appear to be a reference issue.

**Artifact Clue**

In the hinted compiler artifact, the `Lq` tile creates:

```python
LoopSpec(count=sympify("4"))
```

and many ops inside the loop shrink the folded dimension from `16384` to `4096`. The query load looks suspicious:

```python
device_size=[32, 4, 4096, 2, 64]
device_coordinates=[c0, 0, c1, floor(c2/64), Mod(c2, 64)]
```

Here `c0` appears to be the `Hkv` loop but is used directly as the original `Hq` coordinate. For GQA, the original query head should account for both `Hkv` and `G`, roughly `Hq = Hkv * G + g`.

A mismatch diagnostic also supports this: heads `0..3` are correct, while heads `4..31` mismatch, suggesting the folded `[Hkv, G, Lq] -> [Hq, Lq]` mapping is lost or mis-indexed under coarse tiling.

**Suspected Cause**

`coarse_tile`/hint propagation appears to tile `Lq` on the folded `G * Lq` loop, but downstream indexing for the original query view does not preserve the compound named-dim decomposition `["Hkv", "G", "Lq"]`. This causes the tiled kernel to read query heads as if `Hq == Hkv`, dropping the `G` contribution.

### What did you expect to happen?

Tiling or not, it should not create numerical error larger than the tolerance.

### How can we reproduce it?

Tile on a folded dimension.

### Any environmental details we need to know?

No

### Anything else we need to know?

_No response_

### Relevant log output

```Shell

```

### PR DESCRIPTION
**Pull request by cyang49**
Fix gap-dim alignment for folded views

Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

## Summary

Here’s an updated PR description you can paste:

## Summary

This PR fixes two coarse-tiling address correctness issues found while debugging Flash Attention GQA on torch-spyre.

### 1. Fix folded-view gap-dim alignment

The first issue was caused by folded views where `align_tensors()` inserted a physical gap dimension with `coords=0`. That gap dimension could later receive `backGap` assignment in the wrong physical position, causing incorrect address calculations after view/fold transformations.

This PR generalizes the alignment logic to coalesce zero-coordinate physical gap dimensions when they are exactly foldable into neighboring dimensions. This avoids assigning `backGap` to a synthetic gap dimension that does not correspond to a real logical coordinate.

### 2. Fix preserved-loop affine strides for `per_tile_fixed` tensors

A second issue appeared when running with `UNROLL_LOOPS=0`. The unrolled path correctly treats `per_tile_fixed=True` tensors as tile-local scratch and keeps their allocation fixed across loop iterations. However, the preserved-loop symbolic-address path still generated affine address advances for those tensors.

That produced bundle MLIR like:

```mlir
%addr = affine.apply #map(%i_0)[%pool_addr_0]
```

for a pool scratch tensor that should have stayed at `%pool_addr_0`.

~~This PR threads `per_tile_fixed` as codegen-side metadata from `OpSpec` into `generate_sdsc()` and suppresses affine tile strides for those arg indices while still registering their base address normally. `per_tile_fixed` is intentionally not added to `SDSCArgs`, since it is not an SDSC tensor field.~~

Issue 2 is addressed in #3228 

#### Which issue(s) this PR is related to:

Fixes #3098

### CODE REVIEW

**Review by dgrove-oss — CHANGES_REQUESTED** [review]
The second fix from this PR was handled in #3228. 

@tardieu is evaluating the first fix.  We are leaving this PR open while he is doing that, but I'm marking as request changes to prevent an accidental merge.

**Comment by cyang49** [comment]
Closing - not the right fix.
